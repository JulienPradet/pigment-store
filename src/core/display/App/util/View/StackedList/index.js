import React from 'react'
import styles from './stacked-list.m.css'

export const Container = ({children}) => <div className={styles.container}>
  {children}
</div>
Container.displayName = 'StackedList.Container'

export const Row = ({children}) => <div className={styles.row}>
  {children}
</div>
Row.displayName = 'StackedList.Row'

export const Item = ({children}) => <div className={styles.item}>
  {children}
</div>
Item.displayName = 'StackedList.Item'
