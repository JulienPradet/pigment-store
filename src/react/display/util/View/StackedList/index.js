import React from 'react'
import styles from './stacked-list.css'

export const Container = ({children}) => <div className={styles.container}>
  {children}
</div>

export const Row = ({children}) => <div className={styles.row}>
  {children}
</div>

export const Item = ({children}) => <div className={styles.item}>
  {children}
</div>
