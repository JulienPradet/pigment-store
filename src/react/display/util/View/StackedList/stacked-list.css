.container {
  display: block;
  padding: 0;
  margin: 0;
}
.container .container:before {
  content: '>';
  display: block;
  float: left;
  margin-left: -1em;
  margin-right: 1em;
}

.row {
  display: flex;
  padding: 0;
  margin: 0 0 0 0.5em;
}

.item {
  margin: 0 1em;
}

.item:first-child {
  margin-left: 0;
}
.item:last-child {
  margin-right: 0;
}
