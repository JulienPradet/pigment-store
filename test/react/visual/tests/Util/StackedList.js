import React from 'react'
import PigmentStore from '../../../../../dist/index'
import {Container, Row, Item} from '../../../../../src/core/display/App/util/View/StackedList/index'

export default PigmentStore.React.describe('StackedList', Container)
  .featureJsx(
    'Default',
    <Container>
      <Row>
        <Item>
          <a href='/parent'>Parent</a>
        </Item>
        <Item>
          <Container>
            <Row>
              <Item>
                <a href='/parent/child1'>Child 1</a>
              </Item>
            </Row>
            <Row>
              <Item>
                <a href='/parent/child2'>Child 2</a>
              </Item>
            </Row>
            <Row>
              <Item>
                <a href='/parent/child3'>Child 3</a>
              </Item>
            </Row>
          </Container>
        </Item>
      </Row>
    </Container>
  )
  .featureJsx(
    'With many levels of descendance',
    <Container>
      <Row>
        <Item>
          <a href=''>#1</a>
        </Item>
        <Item>
          <Container>
            <Row>
              <Item>
                <a href=''>#11</a>
              </Item>
            </Row>
            <Row>
              <Item>
                <a href=''>#12</a>
              </Item>
              <Item>
                <Container>
                  <Row>
                    <Item>
                      <a href=''>#121</a>
                    </Item>
                    <Item>
                      <Container>
                        <Row>
                          <Item>
                            <a href=''>#1211</a>
                          </Item>
                        </Row>
                        <Row>
                          <Item>
                            <a href=''>#1212</a>
                          </Item>
                        </Row>
                      </Container>
                    </Item>
                  </Row>
                  <Row>
                    <Item>
                      <a href=''>#122</a>
                    </Item>
                  </Row>
                </Container>
              </Item>
            </Row>
            <Row>
              <Item>
                <a href=''>#13</a>
              </Item>
            </Row>
          </Container>
        </Item>
      </Row>
      <Row>
        <Item>
          <a href=''>#2</a>
        </Item>
        <Item>
          <Container>
            <Row>
              <Item>
                <a href=''>#21</a>
              </Item>
            </Row>
            <Row>
              <Item>
                <a href=''>#21</a>
              </Item>
            </Row>
          </Container>
        </Item>
      </Row>
    </Container>
  )
