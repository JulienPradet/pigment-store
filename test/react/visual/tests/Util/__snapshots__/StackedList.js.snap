exports[`StackedList Default 1`] = `
<div
  className="container">
  <div
    className="row">
    <div
      className="item">
      <a
        href="/parent">
        Parent
      </a>
    </div>
    <div
      className="item">
      <div
        className="container">
        <div
          className="row">
          <div
            className="item">
            <a
              href="/parent/child1">
              Child 1
            </a>
          </div>
        </div>
        <div
          className="row">
          <div
            className="item">
            <a
              href="/parent/child2">
              Child 2
            </a>
          </div>
        </div>
        <div
          className="row">
          <div
            className="item">
            <a
              href="/parent/child3">
              Child 3
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`StackedList With many levels of descendance 1`] = `
<div
  className="container">
  <div
    className="row">
    <div
      className="item">
      <a
        href="">
        #1
      </a>
    </div>
    <div
      className="item">
      <div
        className="container">
        <div
          className="row">
          <div
            className="item">
            <a
              href="">
              #11
            </a>
          </div>
        </div>
        <div
          className="row">
          <div
            className="item">
            <a
              href="">
              #12
            </a>
          </div>
          <div
            className="item">
            <div
              className="container">
              <div
                className="row">
                <div
                  className="item">
                  <a
                    href="">
                    #121
                  </a>
                </div>
                <div
                  className="item">
                  <div
                    className="container">
                    <div
                      className="row">
                      <div
                        className="item">
                        <a
                          href="">
                          #1211
                        </a>
                      </div>
                    </div>
                    <div
                      className="row">
                      <div
                        className="item">
                        <a
                          href="">
                          #1212
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="row">
                <div
                  className="item">
                  <a
                    href="">
                    #122
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="row">
          <div
            className="item">
            <a
              href="">
              #13
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="item">
      <a
        href="">
        #2
      </a>
    </div>
    <div
      className="item">
      <div
        className="container">
        <div
          className="row">
          <div
            className="item">
            <a
              href="">
              #21
            </a>
          </div>
        </div>
        <div
          className="row">
          <div
            className="item">
            <a
              href="">
              #21
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
